lib/src/parser.c

```diff
@@ -482,7 +482,7 @@ static Subtree ts_parser__lex(
       current_position.extent.column
     );
     ts_lexer_start(&self->lexer);
-    bool found_token = self->language->lex_fn(&self->lexer.data, lex_mode.lex_state);
+    bool found_token = self->language->lex_fn(&self->lexer.data, lex_mode.lex_state, &self->lexer.accepted_eof);
```

Comment:
I'm not sure that such approach for the implementation is correct, it looks like you created a parallel set of eof states to lex_states and their intersection in a generated parser.c. I thought implementation of the eof() would just generate additional lex states by using existing mechanism, so for some rules the would be generated aditional eof checks and on possitive values transitions to an  aditional state with the aux_sym_eof acceptance.

Reply:
Not exactly, another solution can be to use the field in the TSLexer instead of Lexer, but since that's exposed in the scanner it's maybe better not to.

The issue is we need to know if `aux_sym_eof` was already accepted once before (without advancing), and if so, then the builtin eof sym can be accepted

Reply:
I don't think it should matter whether or not the EOF token was accepted already. Conceptually, it seems like multiple layers of rules should be able to match on `EOF`. Is this intended for preventing infinite loops?

Reply:
Yeah it would endlessly accept `aux_sym_eof` without this check and the parsing would never end.